docs/proposals/004-endpoint-picker-protocol/README.md

```diff
@@ -9,27 +9,55 @@ This doc defines the protocol between the EPP and the proxy (e.g, Envoy).
 The EPP MUST implement the Envoy
 [external processing service](https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/ext_proc/v3/external_processor) protocol.
 
+## Endpoint Subset
+For each HTTP request, the proxy CAN communicate the subset of endpoints the EPP MUST pick from by setting an unstructured entry in the [filter metadata](https://github.com/envoyproxy/go-control-plane/blob/63a55395d7a39a8d43dcc7acc3d05e4cae7eb7a2/envoy/config/core/v3/base.pb.go#L819) field of the ext-proc request. The metadata entry for the subset list MUST be wrapped with an outer key (which represents the metadata namespace) with a default of `envoy.lb.subset_hint`.
+
+```go
+filterMetadata: {
+  "envoy.lb.subset_hint" {
+     "x-gateway-destination-endpoint-subset": [<ip:port>, <ip:port>, ...]
+  }
+}
+```
+
+If the key `x-gateway-destination-endpoint-subset` is set, the EPP MUST only select endpoints from the specified list. If none of the endpoints in the list is eligible or the list is empty, then the EPP MUST return a 429 status code. If the EPP does not select from the list, then this leads to unpredictable behavior.
```

Comment:
Why the 429 status code? 

IMO, the EPP should return an empty pick to allow the proxy's fallback policy to pick in this scenario.

Reply:
This is the current behavior in EPP for the case when there are no ready endpoints, we return 429, so this is being consistent with that.

> IMO, the EPP should return an empty pick to allow the proxy's fallback policy to pick in this scenario.

In all cases? Even when`x-gateway-destination-endpoint-subset` is not set and epp can't find a ready endpoint? How would that work with original_dst cluster that we currently have in the guide?

Reply:
Your other response clarified the conditions under which the 429 is being issued -- with that understanding, the proposed behavior LGTM, but I suggest considering returning 503s as well per my other comment.

Reply:
The status code needs to be updated here as well; might be simpler to reference the other section.

Reply:
EPP is a gRPC service. How can it return 429?

Reply:
Clarified that the status is for immediate response